oauth2client/client.py

```diff
@@ -999,21 +999,17 @@ def _detect_gce_environment():
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
-    connection = six.moves.http_client.HTTPConnection(
-        _GCE_METADATA_HOST, timeout=GCE_METADATA_TIMEOUT)
-
+    http = transport.get_http_object(timeout=GCE_METADATA_TIMEOUT)
```

Comment:
It is "possible", though I'm not sure how it would work with existing `httplib2.Http` objects passed in.

We could file an issue and then follow up after / during redesign? (Now that the quarantine is essentially done.)